### PR TITLE
Add rnix-lsp

### PR DESCRIPTION
**Pull request by mjlbach**
This add rnix-lsp, an lsp that supports nix. I added a basic cargo install, however, I know there are currently discussions about removing this.

### CODE REVIEW

File: lua/nvim_lsp/rnix.lua [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    settings = {
+    };
+    on_new_config = function(config)
+      installer.configure(config)
+    end;
+    init_options = {
+    };
+  };
+  -- on_new_config = function(new_config) end;
+  -- on_attach = function(client, bufnr) end;
```
Comment (h-michael) [inline lua/nvim_lsp/rnix.lua:67]:
```suggestion
```

File: lua/nvim_lsp/rnix.lua [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  docs = {
+    description = [[
+https://github.com/nix-community/rnix-lsp
+
+A language server for Nix providing basic completion and formatting via nixpkgs-fmt.
```
Comment (h-michael) [inline lua/nvim_lsp/rnix.lua:72]: Would you add a short description how to install manually?

File: lua/nvim_lsp/rnix.lua [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  function X.install()
+    local install_info = X.info()
+    if install_info.is_installed then
+      print(name, "is already installed")
+      return
+    end
+    if not (util.has_bins("curl")) then
+      error('Need "curl" to install this.')
+      return
+    end
```
[the remark this answers is not in the document]
Comment (mjlbach) [reply lua/nvim_lsp/rnix.lua]: Fixed

File: lua/nvim_lsp/rnix.lua (hunk 1 above)
Comment (mjlbach) [reply lua/nvim_lsp/rnix.lua]: Fixed

File: lua/nvim_lsp/rnix.lua (hunk 2 above)
Comment (mjlbach) [reply lua/nvim_lsp/rnix.lua:66]: Added, including a note for using with nix.

File: lua/nvim_lsp/rnix.lua [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  end
+  return X
+end
+
+local installer = make_installer()
+
+configs[name] = {
+
+  default_config = {
```
Comment (h-michael) [inline lua/nvim_lsp/rnix.lua:49]: I think it is better to define cmd for those who are not using the installer.
maybe
```suggestion
  default_config = {
    cmd = {"rnix-lsp"};
```

Comment (mjlbach) [reply lua/nvim_lsp/rnix.lua:49]: Done

**Pull request merged**